drivers/gpio/gpio_ite_it8801.c

```diff
@@ -0,0 +1,486 @@
+		return;
+	}
+
+	gpio_it8801_port_get_raw(dev, &value);
+	interrupted_pins_level |= (value & data->level_high_trig);
+	interrupted_pins_level |= ((~value) & data->level_low_trig);
+
+	if ((isr_val & ier_val) || interrupted_pins_level) {
+		/* Clear pending interrupt */
+		ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_gpisr, isr_val);
+		if (ret) {
+			LOG_ERR("Failed to clear GPIO interrupt (ret %d)", ret);
+			return;
+		}
+
+		gpio_fire_callbacks(&data->callbacks, dev, isr_val);
+		/* Level trigger handler again */
+		it8801_gpio_alert_handler(dev);
+	}
+}
+
+static int gpio_it8801_pin_interrupt_configure(const struct device *dev, gpio_pin_t pin,
+					       enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	struct gpio_it8801_data *data = dev->data;
+	int ret;
+	uint8_t reg_gpcr = config->reg_gpcr + pin;
+	uint8_t new_value = 0, control;
+	uint8_t mask = BIT(pin);
+
+	if (ioex_check_is_not_valid(dev, pin)) {
+		return -ENOTSUP;
+	}
+
+	/* Disable irq before configuring it  */
+	ret = i2c_reg_update_byte_dt(&config->i2c_dev, config->reg_gpier, mask, new_value);
+	if (ret) {
+		LOG_ERR("Failed to disable irq (ret %d)", ret);
+		return ret;
+	}
+
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		data->level_high_trig &= ~BIT(pin);
+		data->level_low_trig &= ~BIT(pin);
+
+		return ret;
+	}
+
+	/* Set input pin */
+	ret = i2c_reg_update_byte_dt(&config->i2c_dev, reg_gpcr, IT8801_GPIODIR, 0);
+	if (ret) {
+		LOG_ERR("Failed to set input pin (ret %d)", ret);
+		return ret;
+	}
+
+	/* Clear trigger type */
+	ret = i2c_reg_update_byte_dt(&config->i2c_dev, reg_gpcr, GENMASK(4, 3), 0);
+	if (ret) {
+		LOG_ERR("Failed to clear trigger type (ret %d)", ret);
+		return ret;
+	}
+
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, reg_gpcr, &control);
+	if (ret) {
+		LOG_ERR("Failed to read gpio control (ret %d)", ret);
+		return ret;
+	}
+
+	if (mode == GPIO_INT_MODE_EDGE) {
+		/* Set edge trigger */
+		if ((trig & GPIO_INT_TRIG_BOTH) == GPIO_INT_TRIG_BOTH) {
+			control |= IT8801_GPIOIOT_INT_FALL | IT8801_GPIOIOT_INT_RISE;
+		} else if (trig & GPIO_INT_TRIG_LOW) {
+			control |= IT8801_GPIOIOT_INT_FALL;
+		} else if (trig & GPIO_INT_TRIG_HIGH) {
+			control |= IT8801_GPIOIOT_INT_RISE;
+		} else {
+			LOG_ERR("Invalid interrupt trigger type %d", trig);
+			return -EINVAL;
+		}
+	} else if (mode == GPIO_INT_MODE_LEVEL) {
+		/* Set level trigger */
+		if (trig & GPIO_INT_TRIG_LOW) {
+			control &= ~IT8801_GPIOPOL;
+			data->level_low_trig |= BIT(pin);
+			data->level_high_trig &= ~BIT(pin);
+		} else {
+			control |= IT8801_GPIOPOL;
+			data->level_high_trig |= BIT(pin);
+			data->level_low_trig &= ~BIT(pin);
+		}
+	}
+
+	/* Set control value */
+	ret = i2c_reg_write_byte_dt(&config->i2c_dev, reg_gpcr, control);
+	if (ret) {
+		LOG_ERR("Failed to write trigger state (ret %d)", ret);
+		return ret;
+	}
+
+	/* Clear pending interrupt */
+	new_value = mask;
```

Comment:
can you use mask directly? (if you use the `0` above)

Reply:
Done.